pkg/cloud/api/path.go

```diff
@@ -71,6 +79,26 @@ func (p Path) Equal(other Path) bool {
 	return true
 }
 
+// Similar returns true if other is the same path.
+// For arrays this function does not check indices.
+func (p Path) Similar(other Path) bool {
```

Comment:
How is this method used?

Reply:
I think this function should be used in `fieldTrait` function.
This function checks is search path is in fieldTraits array where we define fields types. The check is done by HasPrefix and IMO it's not good because for structure types we cannot define custom "rules" per fields if we defined rule for structure itself. 
The example :
We have structs:
* TCPRout {
bs BackendService // optional
} 
* BackendService {
name Name // required
rate Rate //optional
}  
We have a TCP route struct which uses BackendService struct as an optional field. BackendService has some optional and required fields like Name. 
In current solution if we are using HAsPrefix to check 2 paths and we define that BackendService is optional field then for path tcproute.backendService.Name we might match the parent tcproute.backendService which is marked as optional field not the field itself. 

And in this function `Equal` will not work because of the exact matching. I. not sure if we need Equal and Similar or just Similar (Equal with logic change to support AnyArrayIndex)  but for sure we need to match paths for arrays :)

Reply:
Can you give a couple of examples with all of the values filled in (maybe as some unit test cases). I'm not 100% following your example (this is probably an issue w/ my understanding).

Reply:
Is this resolved by sorting the trait matching to prefer matching the longest prefix?

Reply:
> Is this resolved by sorting the trait matching to prefer matching the longest prefix?

yes this could be but right now by default we have required fields and we set OutputOnly and AllowZeroValue for some of them. With this approach we would have to define traits for all fields and current implementation with string comparison might be slow (there is a TODO to change this to tree)

Reply:
But also for traits we don't want prefix match but path match right?

Reply:
I think I understand now that you want to keep .Equal to be absolute == and have a separate operation that is interpret match.